src/rulesets/Base/array.jl

```diff
@@ -527,3 +527,42 @@ function _extrema_dims(x, dims)
     end
     return y, extrema_pullback_dims
 end
+
+#####
+##### `pmap`
+#####
+
+using UUIDs # of course we don't want this;
+            # what is a better way of uniquely identifying function calls from the forward pass?
+
+save_backs = Dict() # could this be a stack instead?
+
+# Now that there is a backwards rule for zip (albeit only in Zygote),
+# it should be fine to deal with only a single collection c
+function rrule(config::RuleConfig{>:HasReverseMode}, ::typeof(pmap), f, p::AbstractWorkerPool, c; kwargs...)
+    key = uuid1()
+    function forw(i, a)
+        y, back = rrule_via_ad(config, f, a)
+        save_backs[(key,i)] = back;
```

Comment:
Is the claim that you only want to hold on to the backwards pass? That can probably be done with `Zygote._pullback`

Reply:
I did actually originally write this as a Zygote rule! See this gist here: https://gist.github.com/gaurav-arya/9f78bd49a65a3b1219a1f21cf87fe722